signac/_synced_collections/backends/collection_json.py

```diff
@@ -227,6 +251,7 @@ def _lock_id(self):
         "registry",
         # These keys are specific to the JSON backend.
         "_filename",
+        "filename",
```

Comment:
Is this a sneaky bug fix? A property with the same name existed before, right?

Reply:
Yes, the property existed but was unused. I must have originally just added it for completeness. The bug fix is actually needed now since we are making use of the property, otherwise I would not have noticed.

Reply:
I lied, this wasn't necessary before because the property was only gettable, not settable. The `__getattr__` override is only triggered when the attribute already exists (unlike `__setattr__`), so a property without a setter avoids this problem unless it is also accessed directly in the implementation of `__getitem__` (which is really only `_data`).